### ISSUE TITLE
rpredM for subalgebras

### ISSUE THREAD

**Issue by Tragicus** [math-comp/math-comp#996]
The mulr_closed predicate asks for the subset to contain 1, which is not the case for subalgebras ({aspace _} in field/falgebra.v). Yet these are closed under multiplication and I would like to use rpredM on polynomials with coefficients on a subalgebra. Would it be suitable to make mulrPred depend on mulr_2closed instead of mulr_closed (possibly adding a predicate for what is currently mulr_closed), or at least add a polyOver_mulr_2closed in poly.v ?

**Comment by proux01** [comment]
I agree. You can try adding an intermediate structure in the closedness hierarchy but that'll likely be painful in the current mathcomp (many thing might break). That would be easy in the hierarchy-builder branch though (future mathcomp-2).
Adding a `polyOver_mulr_2closed` in `poly.v` might be a reasonnable short term solution. A PR for any of the three solutions is welcome (but don't spend too much time on the first one).
